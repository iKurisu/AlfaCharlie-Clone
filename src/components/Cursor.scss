.cursor {
  width: 48px;
  height: 48px;
  position: fixed;
  top: -24px;
  left: -24px;
  z-index: 120;
  pointer-events: none;
  transition: 150ms transform ease;
}

.scroll-cursor {
  position: absolute;
  width: 100%;
}

.cursor .-show {
  opacity: 1;
}

.cursor > div:not(.-show) {
  opacity: 0;
}

.cursor > div {
  transition: opacity 0.5s ease;
}
