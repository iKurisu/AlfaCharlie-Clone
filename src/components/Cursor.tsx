import React, { useRef, useEffect } from "react";
import ScrollCursor from "./cursor/Scroll";
import MenuCursor from "./cursor/Menu";
import useMouseFollow from "hooks/useMouseFollow";
import "./Cursor.scss";

const Cursor = (): JSX.Element => {
  const cursor = useRef(null);

  const [followMouse, stopFollowing] = useMouseFollow(cursor);

  useEffect((): (() => void) => {
    followMouse();
    return (): void => stopFollowing();
  }, []);

  return (
    <div className="cursor" ref={cursor}>
      <ScrollCursor />
      <MenuCursor />
    </div>
  );
};

export default Cursor;
