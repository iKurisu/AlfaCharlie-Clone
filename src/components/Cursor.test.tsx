import React from "react";
import renderer from "react-test-renderer";
import Cursor from "./Cursor";

describe("cursor", (): void => {
  it("renders correctly", (): void => {
    const cursor = renderer.create(<Cursor />);

    expect(cursor).toMatchSnapshot();
  });
});
