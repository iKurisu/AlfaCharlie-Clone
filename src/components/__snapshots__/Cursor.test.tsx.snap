// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cursor renders correctly 1`] = `
<div
  className="cursor"
>
  <div
    className="scroll-cursor"
  >
    <div
      className="cursor-outline"
    >
      <svg
        viewBox="0 0 72 72"
        x="0"
        y="0"
      >
        <circle
          cx="36"
          cy="36"
          r="30"
        />
      </svg>
    </div>
    <div
      className="cursor-progress"
    >
      <svg
        viewBox="0 0 72 72"
        x="0"
        y="0"
      >
        <circle
          cx="36"
          cy="36"
          r="30"
          style={
            Object {
              "strokeDashoffset": 200,
            }
          }
        />
      </svg>
    </div>
  </div>
  <div
    className="menu-cursor"
  >
    <div
      className="cursor-outline"
    >
      <svg
        viewBox="0 0 72 72"
        x="0"
        y="0"
      >
        <circle
          cx="36"
          cy="36"
          r="30"
        />
      </svg>
    </div>
    <div
      className="cursor-progress"
    >
      <svg
        viewBox="0 0 72 72"
        x="0"
        y="0"
      >
        <circle
          cx="36"
          cy="36"
          r="30"
          style={
            Object {
              "strokeDashoffset": 200,
            }
          }
        />
      </svg>
    </div>
  </div>
</div>
`;
